version: '3.8'

networks:
  pq_network:
    driver: bridge

services:
  cache:
    image: redis:6.2-alpine
    restart: on-failure
    ports:
      - '6379:6379'
    expose:
      - '6379'
    command: redis-server --save 20 1 --loglevel warning
    volumes:
      - cache:/data
    networks:
      - pq_network

  frontend:
    build:
      context: .
      args:
        service_name: frontend
      dockerfile: ./Dockerfile
    ports:
      - "8080:8080"
    environment:
      DB_HOST: cache
      DB_PORT: 6379
    depends_on:
      - cache
    restart: on-failure
    networks:
      - pq_network
  file-consumer:
    build:
      context: .
      args:
        service_name: fileconsumer
      dockerfile: ./Dockerfile
    environment:
      DB_HOST: cache
      DB_PORT: 6379
    depends_on:
      - cache
    restart: on-failure
    networks:
      - pq_network
  vendor-api-consumer:
    build:
      context: .
      args:
        service_name: vendorapiconsumer
      dockerfile: ./Dockerfile
    environment:
      DB_HOST: cache
      DB_PORT: 6379
    depends_on:
      - cache
    restart: on-failure
    networks:
      - pq_network
  snowflake-api-consumer:
    build:
      context: .
      args:
        service_name: snowflakeconsumer
      dockerfile: ./Dockerfile
    environment:
      DB_HOST: cache
      DB_PORT: 6379
    depends_on:
      - cache
    restart: on-failure
    networks:
      - pq_network

volumes:
  cache:
    driver: local